src/kvstore/Listener.cpp

```diff
@@ -224,17 +224,31 @@ std::pair<int64_t, int64_t> Listener::commitSnapshot(const std::vector<std::stri
                                                      LogID committedLogId,
                                                      TermID committedLogTerm,
                                                      bool finished) {
-    LOG(WARNING) << "Listener snapshot has not implemented yet";
-    UNUSED(committedLogId); UNUSED(committedLogTerm); UNUSED(finished);
+    LOG(INFO) << idStr_ << "Listener is committing snapshot.";
     int64_t count = 0;
     int64_t size = 0;
+    std::vector<KV> data;
     for (const auto& row : rows) {
         count++;
         size += row.size();
-        // todo(doodle): could decode and apply
+        auto kv = decodeKV(row);
+        data.emplace_back(kv.first, kv.second);
     }
-    return {count, size};
-}
+    if (!apply(data)) {
+        LOG(ERROR) << idStr_ << "Failed to apply data while committing snapshot.";
+        return std::make_pair(0, 0);
+    }
+    LOG(INFO) << idStr_ << "Listener succeed to commit snapshot, rows: " << count;
```

Comment:
Maybe this log could be deleted, because `commitSnapshot` could be executed more than once, only when `finished` is true, snapshot is committed totally.

Reply:
OK, will remove it.